prboom2/src/p_map.c

```diff
@@ -2008,6 +2009,7 @@ dboolean PIT_ChangeSector (mobj_t* thing)
     mo = P_SpawnMobj (thing->x,
                       thing->y,
                       thing->z + thing->height/2, MT_BLOOD);
+    mo->target = thing;
```

Comment:
With the way prboom+'s pointer cleanup works, you should use P_SetTarget instead of editing the target value directly. It probably wouldn't happen under normal constraints that a monster hit with an attack might be removed from memory, but I bet it's possible with dehacked, which could lead to some odd behaviour.

Reply:
Yes, it is indeed possible - and actually, doing this unconditionally may cause demo desyncs - not with the regular blood setup but if you add some code pointers that may work with the target pointer, all bets will be off.
I'd actually try to store the MT_ value somewhere instead of a target reference which may cause unforeseen issues

Reply:
Whoa, I wasn't even aware of `P_SetTarget()`, thanks!

> I'd actually try to store the MT_ value somewhere instead of a target reference which may cause unforeseen issues

I'll see what I can do, but if I can't misuse an existing field in the `mobj_t` struct, then I'll have to introduce something new anywhere else. Stay tuned...